KAZOO-5153: make templates great again

icehess opened the pull request:
* Style Templates
* Fix some missing information on templates
* Bring back some templates to life
* Change to transaction: break it into two different templates(one for success, one for payment problems)
* some macro names in templates are changed (port_*, service_added for example)

jamesaimonetti commented on core/braintree/src/braintree_transaction.erl:
```diff
@@ -494,3 +494,29 @@ json_to_record(JObj) ->
                    ,is_recurring = kz_json:is_true(<<"is_recurring">>, JObj)
       }.
+
+-spec record_to_notification_props(bt_transaction()) -> kz_proplist().
```
Unexported / unused function

icehess replied:
I fixed that last night but didn't commit, it's now pushed

jamesaimonetti commented on core/braintree/src/braintree_transaction.erl:
```diff
@@ -494,3 +494,29 @@ json_to_record(JObj) ->
+      ,{<<"Timestamp">>, kz_time:current_tstamp()}
+      ,{<<"Purchase-Order">>, purchase_order_reason(Transaction)}
+      ,{<<"Currency-Code">>, kz_json:get_value(<<"currency_code">>, Transaction)}
+      ].
```
Missing the closing paren for `props:filter_empty/2`

jamesaimonetti commented on core/braintree/src/braintree_transaction.erl:
```diff
@@ -494,3 +494,29 @@ json_to_record(JObj) ->
+-spec record_to_notification_props(bt_transaction()) -> kz_proplist().
+record_to_notification_props(#bt_transaction{}=BraintreeTransaction) ->
+    Transaction = record_to_json(BraintreeTransaction),
+    RespCode = kz_json:get_value(<<"processor_response_code">>, Transaction, ?CODE_UNKNOWN),
```
Macro doesn't exist unless you include `kazoo_transactions.hrl`

icehess replied:
same as above

jamesaimonetti commented on applications/teletype/src/templates/teletype_service_added.erl:
```diff
@@ -98,32 +107,18 @@ reseller_info_data(DataJObj) ->
-              ,{<<"timezone">>, kz_account:timezone(AccountJObj)}
-              ]
-             )
+            teletype_util:find_account_params(DataJObj, ResellerId)
```
ResellerId is undefined

icehess replied:
fixed

jamesaimonetti commented:
@icehess this does not seem like a PR ready for review (lots of compilation errors). Can you close it and reopen when you're ready to have it reviewed? If you want to see CI reports, you can switch to the branch, check the commits list, and there should be the yellow/green/red icon next to the latest commit that will take you to the tests being run.

jamhed commented:
I like the pr header.

fenollp commented:
While the PR does not mention it, this is still ongoing and maybe not ready for review.
But there's no need to close & reopen PR. There's the chance that someone comes around and deletes the branch or something too.

icehess commented:
Except System alert templates, it's ready. You can merge this PR if test passed, no need for waiting to system_alert since this is only would be sent to system admin and there are lots of data dump in it which is really difficult to style it properly.

I'll return it to at a later time.

fenollp commented:
@icehess some Dialyzer complaints ;)
You should rebase before fixing them though, should remove some of them.

icehess commented:
Dialyzer warnings are not related to changes in this PR, they're also exists on the master too. They're related to `bt_transaction{}` record construction which in most places are violating the declared types.

fenollp commented:
I we merge PRs with failing Dialyzer tests this will impact every PRs. We should fix this

k-anderson commented:
@jamesaimonetti hesaam says he addressed your comments, if so can you close the review?

jamesaimonetti commented on applications/teletype/src/templates/teletype_account_zone_change.erl:
```diff
@@ -82,3 +88,12 @@ process_req(DataJObj) ->
         {'error', Reason} -> teletype_util:send_update(DataJObj, <<"failed">>, Reason)
     end.
+
+-spec zones_data(kz_json:object()) -> api_object().
```
Spec is wrong. Return a JSON object in one clause and a proplist of one tuple in the other.

jamesaimonetti replied:
I assume it needs to be `kz_proplist()`?

icehess replied:
fixed

icehess commented:
@jamesaimonetti @lazedo I don't know that you notice this or not, that I had break up `transaction` templates into two templates. Old transaction templates is still has the same id (`transaction`) with friendly_name set to "Transaction Success", and the second one is `transaction_failed` (friendly_name "Transaction Problem").

Also i changed the transaction notification message headers(removed "Transaction" and added some few necessary field instead). This is the change for both `transaction` and `topup`.

I don't know the impact of this changes, @k-anderson told me that it could be a problem.

Any idea?

icehess commented:
Actually some of the macros in templates was changed (port_* and service_added for example).

Also There were some changes in braintree core library, like removing `refund_id` (I couldn't find any reference to it in Braintree docs), adding descriptor etc...
